test/allow-extended-operators.test.js

```diff
@@ -47,16 +32,198 @@ describe('Model.settings.allowExtendedOperators', () => {
     return new DataSource(connectorSettings);
   }
 
+  function setCustomData() {
+    return {$set: {value: 'changed'}};
+  }
+
+  function onSuccess() {
```

Comment:
Sorry if my comment earlier comment was not clear. This is what I meant:

```js
           return inst.updateAttributes(setCustomData(), {allowExtendedOperators: false})
            .then(function onSuccess() {
              throw new Error('updateAttributes() should have failed.');
            })
            .catch(function onError(err) {
              // ...
            });
```

Extracting a shared function is a great idea though! I find the name `onSuccess` too generic, I would like to see a name that reveals more of the intent, for example `updateShouldHaveFailed`.